types/underscore/underscore-tests.ts

```diff
@@ -12,10 +12,12 @@ _.map({ one: 1, two: 2, three: 3 }, (value, key) => value * 3);
 let plucked: string[] = _.map([{key: 'apples'}, {key: 'oranges'}], 'key');
 
 //var sum = _.reduce([1, 2, 3], (memo, num) => memo + num, 0);    // https://typescript.codeplex.com/workitem/1960
-_.reduce([1, 2, 3], (memo, num) => memo + num); // $ExpectType number
-_.reduce([1, 2, 3], (memo, num) => memo + num, 0); // $ExpectType number
-_([1, 2, 3]).reduce((memo, num) => memo + num, 0); // $ExpectType number
-_.chain([1, 2, 3]).reduce((memo, num) => memo + num, 0).value(); // $ExpectType number
+_.reduce([1, 2, 3], (memo, num) => memo + num); // $ExpectType number | undefined
+_.reduce<_.Dictionary<string>, number>({ 'a': '1', 'b': '2', 'c': '3' }, (memo, numstr) => (+memo) + (+numstr)); // $ExpectType string | number | undefined
+_.reduce({ 'a': '1', 'b': '2', 'c': '3' }, (memo: string | number, numstr) => (+memo) + (+numstr)); // $ExpectType string | number | undefined
+_.reduce([1, 2, 3], (memo, num) => memo + num, 0); // $ExpectType number | undefined
+_([1, 2, 3]).reduce((memo, num) => memo + num, 0); // $ExpectType number | undefined
+_.chain([1, 2, 3]).reduce((memo, num) => memo + num, 0).value(); // $ExpectType number | undefined
```

Comment:
Why expect anything other than `number` in any of these cases?

Reply:
Oh, derp, you're right, when given an explicit memo the result is the memo rather than undefined. I'll remove the undefined possibility from the overloads that still take a memo. Thanks for catching this!